ddtrace/contrib/anthropic/utils.py

```diff
@@ -0,0 +1,284 @@
+        self.__wrapped__.__exit__(exc_type, exc_val, exc_tb)
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        try:
+            chunk = self.__wrapped__.__next__()
+            _loop_handler(self._dd_span, chunk, self._streamed_chunks)
+            return chunk
+        except StopIteration:
+            _process_finished_stream(
+                self._dd_integration,
+                self._dd_span,
+                self._args,
+                self._kwargs,
+                self._streamed_chunks,
+                self._is_message_stream,
+            )
+            self._dd_span.finish()
+            self._dd_integration.metric(self._dd_span, "dist", "request.duration", self._dd_span.duration_ns)
+            raise
+        except Exception:
+            self._dd_span.set_exc_info(*sys.exc_info())
+            self._dd_span.finish()
+            self._dd_integration.metric(self._dd_span, "dist", "request.duration", self._dd_span.duration_ns)
+            raise
+
+    def __stream_text__(self):
+        for chunk in self:
+            if chunk.type == "content_block_delta" and chunk.delta.type == "text_delta":
+                yield chunk.delta.text
+
+
+class TracedOpenAIStreamManager(BaseTracedOpenAIStream):
+    def __enter__(self):
+        self.__wrapped__.__enter__()
+        stream = TracedOpenAIStream(
+            self.__wrapped__._MessageStreamManager__stream,
+            self._dd_integration,
+            self._dd_span,
+            self._args,
+            self._kwargs,
+            is_message_stream=True,
+        )
+        stream.text_stream = stream.__stream_text__()
+        return stream
+
+
+class TracedOpenAIAsyncStream(BaseTracedOpenAIStream):
+    async def __aenter__(self):
+        await self.__wrapped__.__aenter__()
+        return self
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        await self.__wrapped__.__aexit__(exc_type, exc_val, exc_tb)
+
+    def __aiter__(self):
+        return self
+
+    async def __anext__(self):
+        try:
+            chunk = await self.__wrapped__.__anext__()
+            _loop_handler(self._dd_span, chunk, self._streamed_chunks)
+            return chunk
+        except StopAsyncIteration:
+            _process_finished_stream(
+                self._dd_integration,
+                self._dd_span,
+                self._args,
+                self._kwargs,
+                self._streamed_chunks,
+                self._is_message_stream,
+            )
+            self._dd_span.finish()
+            self._dd_integration.metric(self._dd_span, "dist", "request.duration", self._dd_span.duration_ns)
+            raise
+        except Exception:
+            self._dd_span.set_exc_info(*sys.exc_info())
+            self._dd_span.finish()
+            self._dd_integration.metric(self._dd_span, "dist", "request.duration", self._dd_span.duration_ns)
+            raise
+
+
+def _construct_unaccumulated_messages(streamed_chunks):
```

Comment:
Will review this further once I get more clarification on my above comments, thanks

Reply:
I will have better comments in my newest commit. This is basically for messages where the user did not use any streaming helper methods, and so a message will have to be built up by looking at each streamed chunk.

Reply:
I've updated this code to be easier to follow.